src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -688,6 +712,23 @@ private void RenderBufferedAttributeValueAccessor(CSharpCodeWriter writer)
             }
         }
 
+        private static bool ContainsChildContent(IList<Chunk> children)
+        {
+            foreach (var child in children)
+            {
+                if (!(child is ParentChunk) || child is TagHelperChunk)
+                {
+                    return true;
+                }
+                else // Non-TagHelper ParentChunk
```

Comment:
The `!` above says this `else` is entered only when `child` _is_ a `ParentChunk` or `TagHelperChunk`.